### ISSUE TITLE
Zen Zen-01-W lacks cool / cooling

### ISSUE THREAD

**Issue by JasonSwindle** [Koenkk/zigbee2mqtt#4320]
# Bug Report
## What happened
Device can support Heat / Cool, but mqtt (and HASS) can only support heat

## What did you expect to happen
The device to be able to do more than heat.

## How to reproduce it (minimal and precise)
Docs and HASS MQTT discover both only have HEAT mode.

## Debug Info
Zigbee2MQTT version: 1.14.4
Adapter hardware: LAUNCHXL CC1352P-2
Adapter firmware version: Z-Stack_3.x.0 (commit f4a0d30302b4757771f01420272b6ced89001dbc)

**Comment by JasonSwindle** [comment]
Is there something more I can provide to help move this forward? :)

**Comment by Koenkk** [comment]
@brubaked do you still have this device? Could you help with this issue?

**Comment by brubaked** [comment]
Alright, my setup since I switched to back from smartthings to Zigbee2MQTT only has heating, so I hadn't noticed this before. I can confirm that when the thermostat is configured to support cooling ([page 19 of this document](https://www.homecontrols.com/homecontrols/products/pdfs/ZE-Zen/Zen-Thermostat-Manual.pdf)) the auto discovery only exposes the heating mode. If I manually configure the thermostat in the configuration.yaml file, the heating and cooling functions work as expected.

```yaml
climate:
  - platform: "mqtt"
    name: "Zen Thermostat"
    availability_topic: "zigbee2mqtt/bridge/state"
    temperature_unit: "C"
    min_temp: "10"
    max_temp: "30"
    mode_state_topic: "zigbee2mqtt/<FriendlyName>"
    mode_state_template: "{{ value_json.system_mode }}"
    mode_command_topic: "zigbee2mqtt/<FriendlyName>"
    current_temperature_topic: "zigbee2mqtt/<FriendlyName>"
    current_temperature_template: "{{ value_json.local_temperature }}"
    temp_step: 0.5
    action_topic: "zigbee2mqtt/<FriendlyName>"
    action_template: "{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}"
    modes: 
      - "off"
      - "heat"
      - "cool"
      - "fan only"
    temperature_state_topic: "zigbee2mqtt/<FriendlyName>"
    temperature_state_template: "{{ value_json.occupied_heating_setpoint }}"
    temperature_command_topic: "zigbee2mqtt/<FriendlyName>/set/occupied_heating_setpoint"
```

The debug log during pairing when the thermostat is configured for heating and cooling is below:

```
Zigbee2MQTT:info  2020-09-14 08:03:41: Device '<FriendlyName>' joined
Zigbee2MQTT:info  2020-09-14 08:03:41: Starting interview of '<FriendlyName>'
Zigbee2MQTT:info  2020-09-14 08:03:41: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"<FriendlyName>"},"type":"device_connected"}'
Zigbee2MQTT:info  2020-09-14 08:03:41: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"<FriendlyName>"},"type":"pairing"}'
Zigbee2MQTT:debug 2020-09-14 08:03:41: Device '<FriendlyName>' announced itself
Zigbee2MQTT:info  2020-09-14 08:03:41: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"<FriendlyName>"},"type":"device_announced"}'
Zigbee2MQTT:debug 2020-09-14 08:03:55: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"modelId":"Zen-01"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:55: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"modelId":"Zen-01"}'
Zigbee2MQTT:info  2020-09-14 08:03:55: MQTT publish: topic 'homeassistant/climate/<FriendlyName>/climate/config', payload '{"action_template":"{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}","action_topic":"zigbee2mqtt/<FriendlyName>","availability_topic":"zigbee2mqtt/bridge/state","current_temperature_template":"{{ value_json.local_temperature }}","current_temperature_topic":"zigbee2mqtt/<FriendlyName>","device":{"identifiers":["zigbee2mqtt_<FriendlyName>"],"manufacturer":"Zen","model":"Thermostat (Zen-01-W)","name":"<FriendlyName>","sw_version":"Zigbee2MQTT 1.14.4"},"json_attributes_topic":"zigbee2mqtt/<FriendlyName>","max_temp":"30","min_temp":"10","mode_command_topic":"zigbee2mqtt/<FriendlyName>/set/system_mode","mode_state_template":"{{ value_json.system_mode }}","mode_state_topic":"zigbee2mqtt/<FriendlyName>","modes":["off","auto","heat"],"name":"<FriendlyName>_climate","temp_step":0.5,"temperature_command_topic":"zigbee2mqtt/<FriendlyName>/set/occupied_heating_setpoint","temperature_state_template":"{{ value_json.occupied_heating_setpoint }}","temperature_state_topic":"zigbee2mqtt/<FriendlyName>","temperature_unit":"C","unique_id":"<FriendlyName>_climate_zigbee2mqtt"}'
Zigbee2MQTT:info  2020-09-14 08:03:55: MQTT publish: topic 'homeassistant/sensor/<FriendlyName>/linkquality/config', payload '{"availability_topic":"zigbee2mqtt/bridge/state","device":{"identifiers":["zigbee2mqtt_<FriendlyName>"],"manufacturer":"Zen","model":"Thermostat (Zen-01-W)","name":"<FriendlyName>","sw_version":"Zigbee2MQTT 1.14.4"},"icon":"mdi:signal","json_attributes_topic":"zigbee2mqtt/<FriendlyName>","name":"<FriendlyName>_linkquality","state_topic":"zigbee2mqtt/<FriendlyName>","unique_id":"<FriendlyName>_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
Zigbee2MQTT:debug 2020-09-14 08:03:55: Received MQTT message on 'homeassistant/climate/<FriendlyName>/climate/config' with data '{"action_template":"{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}","action_topic":"zigbee2mqtt/<FriendlyName>","availability_topic":"zigbee2mqtt/bridge/state","current_temperature_template":"{{ value_json.local_temperature }}","current_temperature_topic":"zigbee2mqtt/<FriendlyName>","device":{"identifiers":["zigbee2mqtt_<FriendlyName>"],"manufacturer":"Zen","model":"Thermostat (Zen-01-W)","name":"<FriendlyName>","sw_version":"Zigbee2MQTT 1.14.4"},"json_attributes_topic":"zigbee2mqtt/<FriendlyName>","max_temp":"30","min_temp":"10","mode_command_topic":"zigbee2mqtt/<FriendlyName>/set/system_mode","mode_state_template":"{{ value_json.system_mode }}","mode_state_topic":"zigbee2mqtt/<FriendlyName>","modes":["off","auto","heat"],"name":"<FriendlyName>_climate","temp_step":0.5,"temperature_command_topic":"zigbee2mqtt/<FriendlyName>/set/occupied_heating_setpoint","temperature_state_template":"{{ value_json.occupied_heating_setpoint }}","temperature_state_topic":"zigbee2mqtt/<FriendlyName>","temperature_unit":"C","unique_id":"<FriendlyName>_climate_zigbee2mqtt"}'
Zigbee2MQTT:debug 2020-09-14 08:03:55: Received MQTT message on 'homeassistant/sensor/<FriendlyName>/linkquality/config' with data '{"availability_topic":"zigbee2mqtt/bridge/state","device":{"identifiers":["zigbee2mqtt_<FriendlyName>"],"manufacturer":"Zen","model":"Thermostat (Zen-01-W)","name":"<FriendlyName>","sw_version":"Zigbee2MQTT 1.14.4"},"icon":"mdi:signal","json_attributes_topic":"zigbee2mqtt/<FriendlyName>","name":"<FriendlyName>_linkquality","state_topic":"zigbee2mqtt/<FriendlyName>","unique_id":"<FriendlyName>_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
Zigbee2MQTT:debug 2020-09-14 08:03:55: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"Zen Within"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:55: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"manufacturerName":"Zen Within"}'
Zigbee2MQTT:debug 2020-09-14 08:03:55: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"powerSource":129}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:55: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"powerSource":129}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"zclVersion":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"zclVersion":1}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"appVersion":23}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"appVersion":23}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"stackVersion":2}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"stackVersion":2}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{"hwVersion":2}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{"hwVersion":2}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{}'
Zigbee2MQTT:debug 2020-09-14 08:03:56: Received Zigbee message from '<FriendlyName>', type 'readResponse', cluster 'genBasic', data '{}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2020-09-14 08:03:56: No converter available for 'Zen-01-W' with cluster 'genBasic' and type 'readResponse' and data '{}'
Zigbee2MQTT:info  2020-09-14 08:03:56: Successfully interviewed '<FriendlyName>', device has successfully been paired
Zigbee2MQTT:info  2020-09-14 08:03:56: Device '<FriendlyName>' is supported, identified as: Zen Thermostat (Zen-01-W)
Zigbee2MQTT:info  2020-09-14 08:03:56: Configuring '<FriendlyName>'
Zigbee2MQTT:info  2020-09-14 08:03:56: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Thermostat","friendly_name":"<FriendlyName>","model":"Zen-01-W","supported":true,"vendor":"Zen"},"type":"pairing"}'
Zigbee2MQTT:info  2020-09-14 08:04:08: Successfully configured '<FriendlyName>'
Zigbee2MQTT:debug 2020-09-14 08:04:58: Received Zigbee message from '<FriendlyName>', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":1,"fileVersion":544,"imageType":2,"manufacturerCode":4440}' from endpoint 1 with groupID 0
```

It seems like the thermostat is indeed only reporting that it has `off`, `auto` and `heat` modes available:
```
"modes":["off","auto","heat"]
```

**Comment by JasonSwindle** [comment]
So does this mean it needs to be fixed at zigbee-herdsman level? Or a flaw with the device?

**Comment by JasonSwindle** [comment]
@Koenkk  @brubaked Following up on this. I hate having to poke people. Thank you for the help. :)

**Comment by brubaked** [comment]
@JasonSwindle Sorry! I haven't had time to dive into this deeper and figure out where the lack of feature autodetection lies. In the meantime, using the YAML code I posted above should give you access to cooling control in Home Assistant.

**Comment by JasonSwindle** [comment]
Howdy,

I will give this a test. By `configuration.yaml` do you mean Zigbee2MQTT or HASS? They both have a configuration.yaml file, but I figure it goes into HASS, as it looks like a config for it.

**Comment by JasonSwindle** [comment]
I'm not sure if this is the dev version, or something else, but I can update MQTT but the device never gets the update. I'm seeing this in the log:

```
ERROR: Publish 'set' 'occupied_heating_setpoint' to 'zen-thermostat' failed: 'Error: Write 0x00244600000eaea7/1 hvacThermostat({"occupiedHeatingSetpoint":1600}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null}) failed (Timeout - 56085 - 1 - 10 - 513 - 4 after 10000ms)'
```

**Comment by JasonSwindle** [comment]
@brubaked Using your setup, I can see MQTT is updated, but the unit never updates to the correct temp. Any advice?

**Comment by JasonSwindle** [comment]
Not stale, but cannot get movement. Bump.

**Comment by stegno** [comment]
I.have problem with integration also. Thermostat reporting temp and heating/idle status correctly, but set_temp works only 1 of 5 times and very laggy...I used deconz soft before and it works better with setting occupied temp.

**Comment by JasonSwindle** [comment]
Bring this back, as this is still not working correctly.

**Comment by gunnarsundberg** [comment]
Having this same issue as well

**Comment by JasonSwindle** [comment]
@Koenkk Can we get this re-opened? The device still not work correctly.

**Comment by gunnarsundberg** [comment]
Hey all! I submitted a fix for this to `zigbee-herdsman-converters` and it was merged. I'm also working on getting a cooling setpoint added.

**Comment by JasonSwindle** [comment]
Thank you so much @gunnarsundberg !

**Comment by ctjameson** [comment]
I'm unfortunately having this issue with a Zen stat. Current version of Zigbee2MQTT and experiencing the exact same issues as OP.

**Comment by gunnarsundberg** [comment]
@ctjameson the problem is in `zigbee-herdsman-converters`. There's no cooling setpoint for the Zen thermostat. I am running a custom version that adds a cooling setpoint. I'll submit a PR when I get the chance

**Comment by ctjameson** [comment]
Thanks @gunnarsundberg. How complicated is it for me to do the same until it's been integrated? Would you be willing to assist me with that? If not, that's fine as well. I can just wait.

**Comment by gunnarsundberg** [comment]
I just submitted #4325

**Comment by ctjameson** [comment]
Is it also possible to add a "fan only" setpoint as well? I leverage the fan in my HVAC very heavily.

**Comment by gunnarsundberg** [comment]
@ctjameson it was merged so you should be able to use the cooling setpoint now! I'll look into the fan only mode

**Comment by ctjameson** [comment]
@gunnarsundberg Apologies for my ignorance regarding how PRs work. I'm running A2M add-in on Home Assistant with a TI coordinator stick. I don't see an update for Z2M currently in the add-on store so I assume I'll need to wait for the next update for your PR to be available to me as a layman? Or is there some type of force update I can run to just pull the zegbee-herdsman-convertors component?

**Comment by andynbaker** [comment]
Hey Z2M gurus! I finally made the jump from Deconz and have repaired all my devices to my conbeeii that is now attached to zigbee2mqtt running as a home-assistant add-on. Everything is working great, but like others here, I can't change the cooling setpoints of my three zen-01-w thermostats. I had this same problem with Deconz and worked with their devs to get that functionality added, so I know the hardware is capable of this function.

It looks like this functionality was merged with the edge branch, so I installed the edge add-on, but I'm still unable to change my cooling setpoints. A couple things I see: 1) there's now this extra temp input in the climate entity, and 2) I now get an error as shown in this screen shot:

![Image 1](https://user-images.githubusercontent.com/53503316/175060155-7a440d08-640d-4be3-931a-32497bb0af6d.png)

Any thoughts?

**Comment by ctjameson** [comment]
@andynbaker I'm glad someone smarter than I when it comes to these is also interested in getting this working. I can't wait to see what happens.

**Comment by ctjameson** [comment]
So I'm on Zen thermostat #2 now. This one is brand new, never used. I can join it to my mesh and connect to the closest coordinator to it (about 6 feet away) but as soon as I put it back on the wall, it's like it loses ability to refresh values at all and just errors out. This is all while the screen is awake and the signal bars show as good. 

I'm still using main line Z2MQTT and not edge, so that may be why but I'm stumped on how to proceed as someone that doesn't know anything about the underlying code.

**Comment by andynbaker** [comment]
I have to walk back my enthusiasm too. I’m seeing now that within a few minutes after pairing it and verifying that it works, they stop responding to the published commands. I too don’t know how to troubleshoot this, so I may sit tight until these changes are published and then create a new issue here.

**Comment by shjryan** [comment]
I am experiencing the same issue as @andynbaker.

**Comment by andynbaker** [comment]
I’ve been using zha since my last post and haven’t looked back. These thermostats work great in that environment.

**Comment by shjryan** [comment]
@andynbaker I've used ZHA to start with, but the thermostat wasn't controllable with ZHA either... That's why I looked into using zigbee2mqtt.

A funny behaviour is that when I make a few changes using HomeKit, I can see them all in the Logbook. However, I don't see the logs in zigbee2mqtt. When I take all the batteries out of the thermostat and reinsert them, and as soon as it connects, all the temp changes get replayed on the thermostat. I really don't know what's happening, whether this is a zigbee2mqtt bug, or the thermostat issue.
